### PR TITLE
Collection of minor fixes for `BlockStorage`

### PR DESCRIPTION
**Pull request by Geod24**
Following last review of `BlockStorage`, there were a few minors issues outstanding.
This collection of commits fixes most of them.

### CODE REVIEW

File: source/agora/common/BlockStorage.d [hunk 1]
```diff
@@ -61,8 +61,10 @@ private struct HashPosition
     size_t              position;
 }
 
-alias IndexHeight = RedBlackTree!(HeightPosition, "(a.height < b.height)");
-alias IndexHash = RedBlackTree!(HashPosition, "(a.hash < b.hash)");
+/// Type of RBTree used for height indexing
+private alias IndexHeight = RedBlackTree!(HeightPosition, "(a.height < b.height)");
```
Comment (AndrejMitrovic) [inline source/agora/common/BlockStorage.d:63]: Could we use lambdas instead?

```D
private alias IndexHeight = RedBlackTree!(HeightPosition, (a, b) => a.height < b.height);
```

It avoids having to mixin strings inside of Phobos, which speeds up compilation and also makes error diagnostics better.

File: source/agora/common/BlockStorage.d [hunk 2]
```diff
@@ -567,7 +565,7 @@ public class BlockStorage
             {
                 idx_file = File(file_name, "rb");
 
-                size_t record_size = (size_t.sizeof * 2 + HASH_SIZE);
+                size_t record_size = (size_t.sizeof * 2 + Hash.sizeof);
```
Comment (AndrejMitrovic) [inline source/agora/common/BlockStorage.d:536]: Could we add a comment why the `size_t.sizeof * 2 + Hash.sizeof`? It's very magic..

File: source/agora/common/BlockStorage.d [hunk 3]
```diff
@@ -36,16 +36,16 @@ import std.stdio;
```
Comment (AndrejMitrovic) [inline source/agora/common/BlockStorage.d:36]: Typo in commit message: `initialzer`

File: source/agora/common/BlockStorage.d [hunk 4]
```diff
@@ -36,16 +36,16 @@ import std.stdio;
...
 
 /// The maximum number of block in one file
-const MFILE_MAX_BLOCK = 100;
+private immutable ulong MFILE_MAX_BLOCK = 100;
 
 /// The map file size
-const MFILE_MAP_SIZE = 640 * 1024;
+private immutable size_t MFILE_MAP_SIZE = 640 * 1024;
 
 private struct HeightPosition
```
Comment (AndrejMitrovic) [inline source/agora/common/BlockStorage.d:50]: This one could use `///`

**Comment by AndrejMitrovic** [comment]
Looks good. Let me know what you think of my comments.

File: source/agora/common/BlockStorage.d (hunk 1 above)
Comment (Geod24) [reply source/agora/common/BlockStorage.d:63]: Well this commit is just about making it private. I actually want to re-explore the use of RBTree here, especially for height.

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by AndrejMitrovic** [comment]
@MukeunKim do these changes look ok to you? You might end up having to rebase your own PR in https://github.com/bpfkorea/agora/pull/158 if https://github.com/bpfkorea/agora/pull/160 is merged.

**Review by MichaelKim20 — APPROVED** [review]

**Review by MichaelKim20 — APPROVED** [review]

**Pull request merged**